src/admin/components/elements/WhereBuilder/index.tsx

```diff
@@ -59,16 +64,28 @@ const WhereBuilder: React.FC<Props> = (props) => {
   const params = useSearchParams();
   const { t, i18n } = useTranslation('general');
 
+  // This handles initializing the where conditions from the search query (URL). That way, if you pass in
+  // query params to the URL, the where conditions will be initialized from those and displayed in the UI.
+  // Example: /admin/collections/posts?where[or][0][and][0][text][equals]=example%20post
   const [conditions, dispatchConditions] = useReducer(reducer, params.where, (whereFromSearch) => {
-    if (modifySearchQuery && validateWhereQuery(whereFromSearch)) {
-      return whereFromSearch.or;
-    }
+    if (modifySearchQuery && whereFromSearch) {
+      if (validateWhereQuery(whereFromSearch)) {
```

Comment:
Validating the where query here seems unnecessary, couldn't you just simply transform it? The transformer function will ensure it's returned in the correct format no matter what. No extra validation needed.

Reply:
@jacobsfletch So, I have tried experimenting with removing that validate and ALWAYS transforming those `where` queries.

The problem here is, that for a valid where query, it would result in additional map operations in the transform step. With that additional validate step, we can skip the unnecessary transform for valid where queries, which will improve the performance. And valid where queries will happen a lot more common than invalid ones anyways.

I'd thus vote for keeping it this way